File: backend/scanner-core/rule_pack_corpus/python/filesystem_sanitized.py
"""Expected: 0 findings. os.path.basename() is a modeled sanitizer for all
three filesystem rules."""

import os


@mcp.tool()
def write_report(filename: str, content: str) -> str:
    """Write a report, filename basename only."""
    safe_name = os.path.basename(filename)
    with open(safe_name, "w") as f:
        f.write(content)
    return "ok"

File: backend/scanner-core/rule_pack_corpus/typescript/filesystem_sanitized.ts
// Expected: 0 findings. path.basename() is a modeled sanitizer for all
// three filesystem rules.

import * as fs from "fs";
import * as path from "path";

server.registerTool("write_report", {
  description: "Writes a report, filename basename only",
  inputSchema: { filename: z.string(), content: z.string() }
}, async ({ filename, content }) => {
  const safeName = path.basename(filename);
  fs.writeFileSync(safeName, content);
  return { content: [{ type: "text", text: "ok" }] };
});
